java/readme.md

```diff
@@ -0,0 +1,213 @@
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+EOF
+```
+
+
+## Get credentials
+Note the username, password and SASL mechanism for the user to authenticate with. Go to the [Security section](acls) to view existing users or create new users. Ensure that the user has ACLs to create, read and write to a topic named `demo-topic`.
+
+
+## Create a topic
+Create a file named `src/main/java/org/example/Main.java` and paste the code below. In the username and password fields, replace the placeholder text with the actual values. Use the SCRAM mechanism that matches the user to authenticate with.
+
+```java title="Main.java"
+package org.example;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.*;
+import org.apache.kafka.clients.producer.*;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.errors.TopicExistsException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+public class Main {
+    static Properties clientProperties() {
+        // Set up producer and consumer properties
+        final String username = "<username>";
+        final String password = "<password>";
+        Properties props = new Properties();
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "<bootstrap-server-address>");
+        props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
+        props.put(SaslConfigs.SASL_MECHANISM, "<SCRAM-SHA-256 or SCRAM-SHA-512>");
+        props.put(SaslConfigs.SASL_JAAS_CONFIG, String.format("org.apache.kafka.common.security.scram.ScramLoginModule required username=\"%s\" password=\"%s\";",
+                username, password));
+        return props;
+    }
+
+    public static void main(String[] args) throws InterruptedException, ExecutionException {
+        createTopic("demo-topic");
+    }
+
+    static void createTopic(String topicName) throws InterruptedException, ExecutionException {
+        try (AdminClient adminClient = AdminClient.create(clientProperties())) {
+            adminClient.createTopics(Collections.singleton(new NewTopic(topicName, Optional.of(1), Optional.empty())))
+                    .all()
+                    .get();
```

Comment:
Report success. E.g. print `demo-topic created`. Otherwise, it ends silently.

Reply:
done (just borrowed the text from the Python example)